### ISSUE TITLE
Portfolio volatility targeting

### ISSUE THREAD

**Issue by jugegp16** [jugegp16/midas#52]
Parent: #51

## Problem

Midas's only portfolio-level risk knobs are `min_cash_pct` and `max_position_pct` (`src/midas/allocator.py:71-91`). Both are static — neither responds to market regime. In a high-vol crash, Midas keeps allocating the same fraction of capital and customers get surprised by drawdowns. (The sample backtest in #29 had a 48.55% max drawdown — on the wrong side of acceptable for most users.)

This issue covers two complementary global gross-exposure controls. They live in the same architectural slot (Phase 3.6 in `Allocator.allocate`) and the implementer needs to think through how they compose — which is why they are bundled here rather than split.

### What pysystemtrade does

Computes predicted portfolio volatility from the weight vector and the instrument covariance matrix, then scales the whole position vector to hit a target annualized vol (e.g. 20%).

### Proposal

Add a post-allocator step:

```python
predicted_vol = sqrt(w @ cov @ w) * sqrt(252)
if predicted_vol > vol_target:
    scale = vol_target / predicted_vol
    targets = {t: w * scale for t, w in targets.items()}
```

Slack from scaling-down goes to cash. Leverage isn't an option for retail equity accounts, so there's no symmetric scale-up — but you can lean further into the highest-conviction names if you want to spend the slack.

### Implementation notes

- Use Ledoit-Wolf shrinkage (`sklearn.covariance.LedoitWolf`) — sample covariance on 5–10 tickers is unstable
- Add `vol_target` to `AllocationConstraints` in `src/midas/models.py`
- The covariance matrix is reused by IDM (#55), so cache it on the result

## Part 2: Realized-drawdown scaling (CPPI-style)

Independent knob, applies regardless of weighting policy:

```yaml
allocator:
  drawdown_scaling:
    enabled: true
    penalty: 1.5     # exposure_scale = max(1 - penalty * current_dd, floor)
    floor: 0.5       # never scale gross below 50% of investable
```

Mechanic: track running peak portfolio value, compute current drawdown, scale `investable` by `max(1 - penalty * current_dd, floor)` *before* the softmax runs. The freed budget becomes deliberate cash reserve. Recovers automatically as portfolio heals.

This is standard CPPI / Kelly-fraction risk control. Bridgewater, Man AHL, Winton, and most institutional trend followers do some form of this. It's universal — works with any weighting policy, any strategy mix, any portfolio shape — and bounds tail risk on portfolios where vol spread makes drawdowns brutal.

The current portfolio value comes from the same place the rebalancer computes it for `current_weights`. Peak tracking lives in the backtest/live engine state, passed into `allocate()` as a single float (`current_drawdown`).

## Composition

Both knobs scale gross exposure but use different signals (predicted vs realized) and apply at different points (post-softmax weight scale vs pre-softmax budget scale). A natural composition:

1. **Drawdown scaling first** — reduces `investable` going into the softmax based on realized drawdown. Acts as a kill switch.
2. **Vol targeting second** — scales the resulting weight vector down further if predicted portfolio vol exceeds target.

This means a user in a 20% drawdown with `penalty=1.5, floor=0.5` enters the softmax with `investable * 0.7`, and the resulting weights may then get further scaled down by vol targeting if the surviving names are still too volatile in aggregate. Both knobs are independently optional — either can be `None` / disabled.

## Acceptance

- [ ] `vol_target` in `AllocationConstraints`, default `None` (off)
- [ ] `drawdown_scaling` (penalty + floor) in `AllocationConstraints`, default off
- [ ] When `vol_target` is set, predicted portfolio vol exceeding target scales weights down
- [ ] When `drawdown_scaling` is enabled, gross exposure reduces during simulated drawdown periods and recovers
- [ ] Both can be enabled simultaneously and compose as documented above
- [ ] Backtest shows reduced drawdown in high-vol regimes vs. baseline
- [ ] Unit test: synthetic high-vol portfolio scales to match `vol_target`
- [ ] Unit test: synthetic drawdown sequence scales `investable` correctly through trough and recovery

## Size

~120 lines + tests (was ~60 for vol-targeting alone; CPPI piece adds ~60)

## Related

- Folded the CPPI / drawdown-aware piece in from #32 (closed as duplicate 2026-04-11)
- #55 IDM reuses the covariance matrix from Part 1
- #53 per-ticker vol scaling is a prerequisite for Part 1 to behave sensibly

**Comment by jugegp16** [comment]
Consolidated into #64.
